### PR TITLE
Remove use of unmodifiable maps

### PR DESCRIPTION
**Pull request by mmoayyed**
Ran into a problem with profile attributes constructed and returned as unmodifiable maps. If the profile is put inside a cache such as Memcached, the unmodifiable map instance cannot be reconstructed when it's fetched from the cache and populated.

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/profile/UserProfile.java [hunk 1]
```diff
@@ -183,14 +183,14 @@ public String getTypedId() {
         return getImmutableAttributeMap(this.authenticationAttributes);
     }
 
-    private Map<String, Object> getImmutableAttributeMap(Map<String, Object> attributeMap) {
+    private static Map<String, Object> getImmutableAttributeMap(Map<String, Object> attributeMap) {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/profile/UserProfile.java]: OK. So let's rename this method as `getAttributeMap` then.

**Comment by leleuj** [comment]
One minor comment. Are you sure the immutable collections are properly registered in the Kyro serializer?

**Comment by mmoayyed** [comment]
Yes, immutables are properly registered with Kryo. It seems like because they are registered, the immutable class is written to the cache. Then when it's read again, Kryp creates an instance of the same immutable structure and attempts to put things into it, which obviously fails. 

Immutables also cause problems if the structure is about to be serialized to JSON for instance. Same problem.

**Comment by leleuj** [comment]
OK. So please rename the `getImmutableAttributeMap` method before I merge.

**Comment by leleuj** [comment]
And the build must be fixed as well (I guess some tests expected immutable collections).

**Comment by leleuj** [comment]
Thanks

**Pull request merged**